### PR TITLE
corrected docstring for Dropout

### LINKED ISSUE #1342: Dropout documentation is incorrect
"Randomly zeroes some of the elements of the input tensor. The elements to zero are randomized on every forward call."

This is incorrect; the function also scales up by 1/(1-p), which the implementation correctly does.

### PR DESCRIPTION
**Pull request by tejaskhot**
fixes #1342

### CODE REVIEW

File: torch/nn/modules/dropout.py [hunk 1]
```diff
@@ -3,9 +3,22 @@
...
+    preventing the co-adaptation of neurons as described in the paper
+    `Improving neural networks by preventing co-adaptation of feature
+    detectors`_.
+
+    Furthermore, the outputs are scaled by a factor of *1/(1-p)* during
+    training. This allows the input to be simply forwarded as-is during
+    evaluation.
+
+    During evaluation, Dropout does nothing more than forward the input such
+    that all elements of the input are considered.
```
Comment (apaszke) [inline torch/nn/modules/dropout.py]: It seems that these two sentences are saying the same. Can you remove them and say "This means that during evaluation the module simply computes an identity function."

Comment (tejaskhot) [inline torch/nn/modules/dropout.py]: Changed it.

File: torch/nn/modules/dropout.py [hunk 2]
```diff
@@ -3,9 +3,18 @@
...
+    tensor with probability *p* using samples from a bernoulli distribution.
     The elements to zero are randomized on every forward call.
 
+    This has proven to be an effective technique for regularization and
+    preventing the co-adaptation of neurons as described in the paper
+    `Improving neural networks by preventing co-adaptation of feature
+    detectors`_.
+
+    Furthermore, the outputs are scaled by a factor of *1/(1-p)* during
+    training. This means that during evaluation the module simply computes
```
Comment (apaszke) [inline torch/nn/modules/dropout.py]: it seems that you didn't finish the sentence?

**Review by apaszke — COMMENTED** [review]
I'm getting these sphinx errors when building the docs:
```
/data/users/apaszke/pt/torch/nn/modules/dropout.py:docstring of torch.nn.Dropout:26: WARNING: malformed hyperlink target.                                                                                   
/data/users/apaszke/pt/torch/nn/modules/dropout.py:docstring of torch.nn.Dropout:27: WARNING: Explicit markup ends without a blank line; unexpected unindent.                                               
/data/users/apaszke/pt/torch/nn/modules/dropout.py:docstring of torch.nn.Dropout:5: ERROR: Unknown target name: "improving neural networks by preventing co-adaptation of feature detectors".               
```

**Review by apaszke — APPROVED** [review]

**Comment by apaszke** [comment]
Thanks!
